smpperf.go

```diff
@@ -125,23 +130,33 @@ func (s *SMPPerf) SendMessages() {
 
 		conn := transceiver.Bind() // make persistent connection.
 		defer transceiver.Close()
+		// make sure connection is alright
 		for c := range conn {
 			if c.Error() == nil {
 				break
 			}
 			log.Println("ERROR: connection failed:", c.Error())
 		}
+		// close connection if Interrupted
 		closeTransceiverOnSignal(transceiver)
 
-		go func() {
+		t := &transceiverConn{
+			transceiver: transceiver,
+			err:         nil,
+		}
+
+		// report error on failed conn and Increment error count
+		go func(transceiverIndex int) {
 			for c := range conn {
+				go func() { transceivers[transceiverIndex].err = c.Error() }()
```

Comment:
I think there's a race condition with this statement and the check of `err` below in the code.